src/commands/createFunctionApp/containerImage/detectDockerfile.ts

```diff
@@ -2,66 +2,98 @@
+    const dockerfiles: Uri[] = await workspace.findFiles(pattern);
+    context.telemetry.properties.dockerfileCount = String(dockerfiles.length);
+
+    if (dockerfiles.length === 0) {
+        context.telemetry.properties.containerizedDockerfileCount = '0';
+        return;
     }
+
+    const useContainerImage: boolean = await promptUseContainerImage(context);
+    context.telemetry.properties.useContainerImage = String(useContainerImage);
+
+    if (!useContainerImage) {
+        return;
+    }
+
+    if (dockerfiles.length === 1) {
+        context.dockerfilePath = dockerfiles[0].fsPath;
+        context.telemetry.properties.containerizedDockerfileCount = await detectFunctionsDockerfile(context.dockerfilePath) ? '1' : '0';
+    } else {
+        context.dockerfilePath = await promptChooseDockerfile(context, dockerfiles);
+    }
+
+    // ensure container apps extension is installed before proceeding
+    await getAzureContainerAppsApi(context);
 }
 
-export async function detectFunctionsDockerfile(file: string): Promise<boolean> {
-    const content = await AzExtFsExtra.readFile(file);
-    const lines: string[] = content.split('\n');
+async function promptUseContainerImage(context: IActionContext): Promise<boolean> {
+    const placeHolder: string = localize('detectedDockerfile', 'Dockerfile detected. What would you like to deploy?');
+    const containerImageButton: MessageItem = { title: localize('containerImage', 'Container Image') };
+    const codeButton: MessageItem = { title: localize('code', 'Code') };
+    const buttons: MessageItem[] = [containerImageButton, codeButton];
+    const result: MessageItem = await context.ui.showWarningMessage(placeHolder, { modal: true }, ...buttons);
+    return result === containerImageButton;
+}
+
+async function promptChooseDockerfile(context: ICreateFunctionAppContext, dockerfiles: Uri[]): Promise<string> {
+    const checkContainerizedDockerfiles: Promise<boolean>[] = dockerfiles.map(d => detectFunctionsDockerfile(d.fsPath));
+
+    let containerizedDockerfileCount: number = 0;
+    const picks: IAzureQuickPickItem<string | undefined>[] = [];
 
-    for (const line of lines) {
-        if (line.includes('mcr.microsoft.com/azure-functions')) {
-            return true;
+    for (const [i, dockerfile] of dockerfiles.entries()) {
+        const isContainerizedDockerfile: boolean = await checkContainerizedDockerfiles[i];
+        if (isContainerizedDockerfile) {
+            containerizedDockerfileCount++;
         }
+
+        const relativeDirectory: string = '.' + path.sep + path.relative(nonNullProp(context, 'rootPath'), dockerfile.fsPath);
+        const functionsImage: string = localize('functionsImage', 'Functions Image');
+
+        picks.push({
+            label: path.basename(dockerfile.fsPath),
+            description: isContainerizedDockerfile ? `${relativeDirectory} (${functionsImage})` : relativeDirectory,
```

Comment:
showing pick description as `(Functions Image)` when the Dockerfile leverages the Azure Functions base image